plenum/server/quota_control.py

```diff
@@ -0,0 +1,46 @@
+
+class QuotaControl:
+    def __init__(self,
+                 max_node_message_count_limit,
+                 max_node_message_size_limit,
+                 max_client_message_count_limit,
+                 max_client_message_size_limit):
+        self._max_node_message_count_limit = max_node_message_count_limit
+        self._max_node_message_size_limit = max_node_message_size_limit
+        self._max_client_message_count_limit = max_client_message_count_limit
+        self._max_client_message_size_limit = max_client_message_size_limit
+        self._node_quota_reached = False
+
+    def received_node_messages(self, count: int, size: int):
+        if count >= self._max_node_message_count_limit:
+            self._node_quota_reached = True
+            return
+
+        if size >= self._max_node_message_size_limit:
+            self._node_quota_reached = True
+            return
+
```

Comment:
May be do like this?)
reached = count >= limit or size >= limit

Reply:
Good catch )